GREEDY/DataManagers/ReceiptCreating.cs

```diff
@@ -0,0 +1,105 @@
+﻿using GREEDY.Data;
+using GREEDY.Models;
+using GREEDY.OCRs;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace GREEDY.DataManagers
+{
+    public class ReceiptCreating : IReceiptCreating
+    {
+        private readonly IOcr _ocr;
+
+        public ReceiptCreating()
+        {
+            _ocr = new EmguOcr();
+            //TODO: !!! need to find better place or create new class to fill all tables data as we need !!!
+            AddDataToShopDataTable();
+        }
+
+        public Receipt FullReceiptCreating(Bitmap image)
+        {
+            var linesOfText = _ocr.ConvertImage(image);
+            var date = GetDateForReceipt(linesOfText);
+            var shop = GetShopFromData(linesOfText);
+
+            return new Receipt
+            {
+                Date = date,
+                Shop = shop,
+                LinesOfText = linesOfText
+            };
+        }
+
+        public string GetDateForReceipt(List<string> linesOfText)
+        {
+            var receiptLinesToString = String.Join(Environment.NewLine, linesOfText);
+            string pattern = @"(\d{4}-\d{2}-\d{2})(\d{2})?";
+            receiptLinesToString = Regex.Replace(receiptLinesToString, @"~", "-");
+
+            Match match = Regex.Match(receiptLinesToString, pattern, RegexOptions.Singleline);
+            if (match.Success)
+            {
+                return match.Groups[1].Value;
+            }
+            else
+            {
+                return DateTime.Today.ToString("yyyy-MM-dd");
+            }
+        }
+
+        private void AddDataToShopDataTable()
+        {
+            dynamic dynJson = JsonConvert.DeserializeObject<List<Shop>>(
+                @"[
+                    {'Name': 'IKI','Adress': '','Subname': 'PALINK'},
+                    {'Name': 'MAXIMA','Adress': '', 'Subname': ''},
+                    {'Name': 'RIMI','Adress': '','Subname': ''},
+                    {'Name': 'LIDL','Adress': '','Subname': ''}
+                ]");
```

Comment:
Why would you hard code it as json only to convert to object? If it is hardcoded you can just create it as object